fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/jobs/recalculateinterestforloan/RecalculateInterestForLoanTasklet.java

```diff
@@ -91,8 +91,8 @@ public RepeatStatus execute(StepContribution contribution, ChunkContext chunkCon
 
     private void recalculateInterest(OfficeData office, int threadPoolSize, int batchSize) {
         final int pageSize = batchSize * threadPoolSize;
-
-        final ExecutorService executorService = Executors.newFixedThreadPool(threadPoolSize);
+        taskExecutor.setCorePoolSize(threadPoolSize);
```

Comment:
Sooo, are we updating the global default task executor configuration? This mustn't be right..

Reply:
a configurable task executor has been made, which is a prototype bean